### ISSUE TITLE
Improper space handling in path in function gef_get_auxiliary_values

### ISSUE THREAD

**Issue by extremecoders-re** [hugsy/gef#204]
### Environment

  * Operating System: Ubuntu 16.04
  * Architecture: Tested on x86-64 (However applicable to all archs)
  * GDB version (including the Python library version):  7.11.1, python 3.5

#### Steps to reproduce

  1. Debug any binary **having spaces in its path**
  2. Invoke the command `castone-disassemble` on an address

#### Observed Results

gef throws an error
```
[!] Command 'capstone-disassemble' failed to execute properly, reason: invalid literal for int() with base 0: '"/home/ec/oae/II/format'
```

### Traces

```
─────────────────────────────── Exception raised ───────────────────────────────
ValueError: invalid literal for int() with base 0: '"/home/ec/oae/II/format'
───────────────────────────── Detailed stacktrace ──────────────────────────────
↳ File "/home/ec/.gdbinit-gef.py", line 2835, in gef_get_auxiliary_values()
    →     res[_type] = int(tmp[-2], base=0) if _type in ("AT_PLATFORM", "AT_EXECFN") else int(tmp[-1], base=0)
↳ File "/home/ec/.gdbinit-gef.py", line 2858, in gef_getpagesize()
    →     auxval = gef_get_auxiliary_values()
↳ File "/home/ec/.gdbinit-gef.py", line 1190, in capstone_disassemble()
    →     code = kwargs.get("code", read_memory(location, gef_getpagesize() - offset - 1))
↳ File "/home/ec/.gdbinit-gef.py", line 5404, in do_invoke()
    →     for insn in capstone_disassemble(location, length, **kwargs):
↳ File "/home/ec/.gdbinit-gef.py", line 1993, in wrapper()
    →     return f(*args, **kwargs)
↳ File "/home/ec/.gdbinit-gef.py", line 3351, in invoke()
    →     self.do_invoke(argv)
```

### My understanding of the problem

The root cause of the bug is due to improper space handling in function `gef_get_auxiliary_values`

https://github.com/hugsy/gef/blob/bc6b070d08fcfaba7febfaa725900c55b8eb8f39/gef.py#L2832-L2836

If the path contains spaces `info auxv` returns something like this.

```
gef➤  info auxv 
...
31   AT_EXECFN   File name of executable   0x7fffffffefd1   "/home/ec/oae/II/format str/formatstr33"
...
```

Note that there is a space in the path. Due to this `tmp = line.split()` splits it into two items.

```
tmp[-1] = "str/formatstr33"
tmp[-2] = "/home/ec/oae/II/format"
```

As a result the following line throws an error
```
 res[_type] = int(tmp[-2], base=0) if _type in ("AT_PLATFORM", "AT_EXECFN") else int(tmp[-1], base=0)
```

since it tries to execute 
```
int("/home/ec/oae/II/format", base=0)
```

**Comment by hugsy** [comment]
TODO:

 - [ ] fix the wrong filename parsing in `gef_get_auxiliary_values`
 - [ ] add regression test for `gef_get_auxiliary_values` testing filename with spaces

**Comment by hugsy** [comment]
Fixed by 6d0fb48.

@extremecoders-re  Can you confirm and close the issue if it's ok ? Thanks

**Comment by extremecoders-re** [comment]
Issue has been tested and confirmed to be fixed! 
Closing.
